File: Java/Cristian Maldonado/BinarySearch.java
import java.util.Scanner;

public class BinarySearch {

	public static void main(String[] args) {
		Scanner sc = new Scanner (System.in);
		int arr[]= {1,2,3,4,5,6};
		int tam = arr.length;
		int numerobuscado = 5;
		int resultado = binarySearch(arr, 0, tam-1, numerobuscado);
		if(resultado == -1) {
			System.out.println("Elemento no encontrado");
		}else {
			System.out.println("Elemento encontrado en el indice "+resultado);
		}
	}
	
	static int binarySearch(int arr[],int inicio, int tamfinal, int numerobuscado) {
		if(tamfinal>=inicio) {
			int medio = inicio+(tamfinal-1)/2; 
			if(arr[medio]== numerobuscado) {
				return medio;
			}
			if(arr[medio]>numerobuscado) {
				return binarySearch(arr, inicio, medio-1, numerobuscado);
			}else {
				return binarySearch(arr, medio+1, tamfinal, numerobuscado);
			}
		}
		return -1;
	}

}

File: Codigos estudiantes por lenguaje/PY/Daniel Alayon/Algoritmos de busqueda/binarySearch.py
def BinarySearch(arr, inicio, fin, numerobuscado):
    if fin >= inicio:
        medio = int(inicio + (fin - 1) / 2)
        if arr[medio] == numerobuscado:
            return medio
        if arr[medio] > numerobuscado:
            return BinarySearch(arr, inicio, medio-1, numerobuscado)
        else:
            return BinarySearch(arr, medio + 1, fin, numerobuscado)
    return -1


arr = [1, 2, 3, 4, 5, 6]

tam = len(arr)
numerobuscado = 5
resultado = int(BinarySearch(arr, 0, tam-1, numerobuscado))

if resultado == -1:
    print("Elemento no encontrado")
else:
    print("Elemento encontrado en la posicion: ", resultado)
